lambda_src/drivers/destination_s3.py

```diff
@@ -82,15 +82,17 @@ def write_to_s3(bucket: Text, filename: Text, content: AnyStr) -> Dict[Text, Any
     )
 
 
-def initialize(destination, batch_id: Text):
+def initialize(destination: Text):
     bucket, prefix = parse_destination_uri(destination)
     content = ''  # We use empty body for creating a folder
-    if not prefix:
-        prefixed_folder_path = f'{DATA_FOLDER_NAME}/{batch_id}/'
-    else:
-        prefixed_folder_path = f'{prefix}/{batch_id}/'
+    prefixed_folder_path = ''
+    if prefix and prefix.count('/') and not prefix.endswith('/'):
+        prefixed_folder_path = '/'.join(prefix.split('/')[:-1]) + '/'
+    elif prefix.endswith('/'):
+        prefixed_folder_path = prefix
 
-    return write_to_s3(bucket, prefixed_folder_path, content)
+    if prefixed_folder_path:
+        return write_to_s3(bucket, prefixed_folder_path, content)
```

Comment:
the return here seems to suggest there's meaning to this response value, but that's misleading because the code that invokes it does not store the return value

Reply:
Ah right, should this then just be made into an invocation by removing the return?